### ISSUE TITLE
SymTest does not allow a List to be mapped into another type

### ISSUE THREAD

**Issue by catchmareck** [microsoft/BosqueLanguage#376]
**Describe the bug**
When I create a List of type T and try to map it into a list of type U the SymTest tool throws a parser error:

```
Parse error -- ["[...]\\BosqueLanguage\\impl\\bin\\core\\symbolic\\list.bsq",415,"Parameter l expected argument of type NSCore::List<NSMain::Sandwich> but got NSCore::List<NSCore::Bool>"]
```

**To Reproduce**
Create a file `a.bsq` and put the following content into it:

```
namespace NSMain;

entity Sandwich {
    field cheese: Bool;
}

entrypoint function main(): List<Bool> {

    let sl = List<Sandwich>@{ Sandwich@{ true }, Sandwich@{ false } };

    return sl.map<Bool>(fn(s) => s.cheese);
}
```

Then, run the SymTest tool

```
node bin/runtimes/symtest/symtest.js a.bsq
```

**Expected behavior**
I expect it to analyze the code and try to find possible failing inputs or return `Verified up to bound -- no errors found!`.

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Bosque Version `master`

**Additional context**
The program compiled without any issues and runs properly. There is a problem with parsing when using the SymTest I guess. The exact same error appears when I use `.joinGroup<U>()` method and I suspect that it fails for any method that is parameterized by type `U` while the List is of type `T`.

**Comment by mrkmarron** [comment]
Just merged in your PR to fix this issue. Thanks again for your contribution.

**Comment by catchmareck** [comment]
I'm afraid the #377 that you just merged does not fix this particular issue. In that PR I fixed the SMT body emitter because it was impossible to run symtest on more complex programs (with custom types, entities, etc).

Now, that it's been merged the issue I described there does not occur anymore BUT this one - #376 - still does occur because it's a separate problem. Therefore, I think it should be reopened.

Sorry for the confusion.
